File: app-sdk/src/main/java/code/sdk/bridge/Bridge.java
package code.sdk.bridge;

import code.util.ByteUtil;
import code.util.NumberUtil;

/**
 * 所有桥接方法的父类，按照方法名进行调用分发，方法实现类请继承实现接口: BridgeInterface
 */
public abstract class Bridge {

    private BridgeInterface mBridgeInterface;

    public Bridge(BridgeInterface bridgeInterface) {
        mBridgeInterface = bridgeInterface;
    }

    /**
     * 处理业务层上层（Cocos或者H5）发送的调用
     *
     * @param request
     * @return
     */
    public abstract String post(String request);

    protected String dispatchRequest(String method, String[] params) {
        if (params == null) {
            return "";
        }
        String result = "";
        switch (method) {
            case "nativeLog":
                if (params.length >= 2) {
                    mBridgeInterface.nativeLog(params[0], params[1]);
                }
                break;
            case "copyText":
                if (params.length >= 1) {
                    mBridgeInterface.copyText(params[0]);
                }
                break;
            case "getCopiedText":
                result = mBridgeInterface.getCopiedText();
                break;
            case "showNativeToast":
                if (params.length >= 1) {
                    mBridgeInterface.showNativeToast(params[0]);
                }
                break;
            case "initAdjustID":
                if (params.length >= 1) {
                    mBridgeInterface.initAdjustID(params[0]);
                }
                break;
            case "trackAdjustEvent":
                if (params.length >= 2) {
                    mBridgeInterface.trackAdjustEvent(params[0], params[1]);
                }
                break;
            case "trackAdjustEventStart":
                if (params.length >= 1) {
                    mBridgeInterface.trackAdjustEventStart(params[0]);
                }
                break;
            case "trackAdjustEventGreeting":
                if (params.length >= 1) {
                    mBridgeInterface.trackAdjustEventGreeting(params[0]);
                }
                break;
            case "trackAdjustEventAccess":
                if (params.length >= 1) {
                    mBridgeInterface.trackAdjustEventAccess(params[0]);
                }
                break;
            case "trackAdjustEventUpdated":
                if (params.length >= 1) {
                    mBridgeInterface.trackAdjustEventUpdated(params[0]);
                }
                break;
            case "getDeviceID":
                result = mBridgeInterface.getDeviceID();
                break;
            case "getDeviceInfoForLighthouse":
                result = mBridgeInterface.getDeviceInfoForLighthouse();
                break;
            case "getSystemVersionCode":
                result = String.valueOf(mBridgeInterface.getSystemVersionCode());
                break;
            case "getClientVersionCode":
                result = String.valueOf(mBridgeInterface.getClientVersionCode());
                break;
            case "getPackageName":
                result = mBridgeInterface.getPackageName();
                break;
            case "getAppName":
                result = mBridgeInterface.getAppName();
                break;
            case "getChannel":
                result = mBridgeInterface.getChannel();
                break;
            case "getBrand":
                result = mBridgeInterface.getBrand();
                break;
            case "saveGameUrl":
                if (params.length >= 1) {
                    mBridgeInterface.saveGameUrl(params[0]);
                }
                break;
            case "saveAccountInfo":
                if (params.length >= 1) {
                    mBridgeInterface.saveAccountInfo(params[0]);
                }
                break;
            case "getAccountInfo":
                result = mBridgeInterface.getAccountInfo();
                break;
            case "getAdjustDeviceID":
                result = mBridgeInterface.getAdjustDeviceID();
                break;
            case "getGoogleADID":
                result = mBridgeInterface.getGoogleADID();
                break;
            case "getIDFA":
                result = mBridgeInterface.getIDFA();
                break;
            case "getReferID":
                result = mBridgeInterface.getReferID();
                break;
            case "getAgentID":
                result = mBridgeInterface.getAgentID();
                break;
            case "setCocosData":
                if (params.length >= 2) {
                    mBridgeInterface.setCocosData(params[0], params[1]);
                }
                break;
            case "getCocosData":
                if (params.length >= 1) {
                    result = mBridgeInterface.getCocosData(params[0]);
                }
                break;
            case "getCocosAllData":
                result = mBridgeInterface.getCocosAllData();
                break;
            case "getLighterHost":
                result = mBridgeInterface.getLighterHost();
                break;
            case "getBridgeVersion":
                result = String.valueOf(mBridgeInterface.getBridgeVersion());
                break;
            case "isFacebookEnable":
                result = String.valueOf(mBridgeInterface.isFacebookEnable());
                break;
            case "getTDTargetCountry":
                result = mBridgeInterface.getTDTargetCountry();
                break;
            case "openUrlByBrowser":
                if (params.length >= 1) {
                    mBridgeInterface.openUrlByBrowser(params[0]);
                }
                break;
            case "openUrlByWebView":
                if (params.length >= 1) {
                    mBridgeInterface.openUrlByWebView(params[0]);
                }
                break;
            case "openApp":
                if (params.length >= 2) {
                    mBridgeInterface.openApp(params[0], params[1]);
                }
                break;
            case "loadUrl":
                if (params.length >= 1) {
                    mBridgeInterface.loadUrl(params[0]);
                }
                break;
            case "goBack":
                mBridgeInterface.goBack();
                break;
            case "close":
                mBridgeInterface.close();
                break;
            case "refresh":
                mBridgeInterface.refresh();
                break;
            case "clearCache":
                mBridgeInterface.clearCache();
                break;
            case "saveImage":
                if (params.length >= 1) {
                    mBridgeInterface.saveImage(params[0]);
                }
                break;
            case "savePromotionMaterial":
                if (params.length >= 1) {
                    mBridgeInterface.savePromotionMaterial(params[0]);
                }
                break;
            case "synthesizePromotionImage":
                if (params.length >= 4) {
                    mBridgeInterface.synthesizePromotionImage(params[0],
                            NumberUtil.parseInt(params[1]),
                            NumberUtil.parseInt(params[2]),
                            NumberUtil.parseInt(params[3])
                    );
                }
                break;
            case "shareUrl":
                if (params.length >= 1) {
                    mBridgeInterface.shareUrl(params[0]);
                }
                break;
            case "loginFacebook":
                mBridgeInterface.loginFacebook();
                break;
            case "logoutFacebook":
                mBridgeInterface.logoutFacebook();
                break;
            case "preloadPromotionImage":
                if (params.length >= 1) {
                    mBridgeInterface.preloadPromotionImage(params[0]);
                }
                break;
            case "shareToWhatsApp":
                if (params.length >= 1) {
                    mBridgeInterface.shareToWhatsApp(params[0]);
                }
                break;
            case "isHttpDnsEnable":
                result = String.valueOf(mBridgeInterface.isHttpDnsEnable());
                break;
            case "httpdns":
                if (params.length >= 1) {
                    result = mBridgeInterface.httpdns(params[0]);
                }
                break;
            case "httpdnsInit":
                if (params.length >= 1) {
                    mBridgeInterface.httpdnsInit(params[0]);
                }
                break;
            case "httpdnsRequestSync":
                if (params.length >= 2) {
                    result = mBridgeInterface.httpdnsRequestSync(params[0], ByteUtil.stringToBytes(params[1]));
                }
                break;
            case "httpdnsRequestAsync":
                if (params.length >= 2) {
                    mBridgeInterface.httpdnsRequestAsync(params[0], ByteUtil.stringToBytes(params[1]));
                }
                break;
            case "httpdnsWsOpen":
                if (params.length >= 1) {
                    mBridgeInterface.httpdnsWsOpen(params[0]);
                }
                break;
            case "httpdnsWsSend":
                if (params.length >= 2) {
                    result = mBridgeInterface.httpdnsWsSend(params[0], ByteUtil.stringToBytes(params[1]));
                }
                break;
            case "httpdnsWsClose":
                if (params.length >= 1) {
                    mBridgeInterface.httpdnsWsClose(params[0]);
                }
                break;
            case "httpdnsWsConnected":
                if (params.length >= 1) {
                    result = mBridgeInterface.httpdnsWsConnected(params[0]);
                }
                break;
            case "getBuildVersion":
                result = mBridgeInterface.getBuildVersion();
                break;
            case "onAnalysisStart":
                if (params.length >= 2) {
                    mBridgeInterface.onAnalysisStart(params[0], NumberUtil.parseLong(params[1]));
                }
                break;
            case "onAnalysisEnd":
                mBridgeInterface.onAnalysisEnd();
                break;
            case "memoryInfo":
                result = mBridgeInterface.memoryInfo();
                break;
            case "isEmulator":
                result = String.valueOf(mBridgeInterface.isEmulator());
                break;
            case "commonData":
                result = mBridgeInterface.commonData();
                break;
            case "exitApp":
                mBridgeInterface.exitApp();
                break;
            case "handleNotification":
                mBridgeInterface.handleNotification();
                break;
            default:
                break;
        }
        return result;
    }

}

File: app-sdk/src/main/java/code/sdk/bridge/Bridge.kt
package code.sdk.bridge

import code.util.ByteUtil.stringToBytes
import code.util.NumberUtil.parseInt
import code.util.NumberUtil.parseLong

/**
 * 所有桥接方法的父类，按照方法名进行调用分发，方法实现类请继承实现接口: BridgeInterface
 */
abstract class Bridge(private val mBridgeInterface: BridgeInterface) {
    /**
     * 处理业务层上层（Cocos或者H5）发送的调用
     *
     * @param request
     * @return
     */
    abstract fun post(request: String?): String?
    protected fun dispatchRequest(method: String?, params: Array<String?>?): String? {
        if (params == null) {
            return ""
        }
        var result:String? = ""
        when (method) {
            "nativeLog" -> {
                if (params.size >= 2) {
                    mBridgeInterface.nativeLog(params[0], params[1])
                }
            }

            "copyText" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.copyText(params[0])
                }
            }

            "getCopiedText" -> {
                result = mBridgeInterface.getCopiedText()
            }

            "showNativeToast" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.showNativeToast(params[0])
                }
            }

            "initAdjustID" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.initAdjustID(params[0])
                }
            }

            "trackAdjustEvent" -> {
                if (params.size >= 2) {
                    mBridgeInterface.trackAdjustEvent(params[0], params[1])
                }
            }

            "trackAdjustEventStart" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.trackAdjustEventStart(params[0])
                }
            }

            "trackAdjustEventGreeting" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.trackAdjustEventGreeting(params[0])
                }
            }

            "trackAdjustEventAccess" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.trackAdjustEventAccess(params[0])
                }
            }

            "trackAdjustEventUpdated" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.trackAdjustEventUpdated(params[0])
                }
            }

            "getDeviceID" -> {
                result = mBridgeInterface.getDeviceID()
            }

            "getDeviceInfoForLighthouse" -> {
                result = mBridgeInterface.getDeviceInfoForLighthouse()
            }

            "getSystemVersionCode" -> {
                result = mBridgeInterface.getSystemVersionCode().toString()
            }

            "getClientVersionCode" -> {
                result = mBridgeInterface.getClientVersionCode().toString()
            }

            "getPackageName" -> {
                result = mBridgeInterface.getPackageName()
            }

            "getAppName" -> {
                result = mBridgeInterface.getAppName()
            }

            "getChannel" -> {
                result = mBridgeInterface.getChannel()
            }

            "getBrand" -> {
                result = mBridgeInterface.getBrand()
            }

            "saveGameUrl" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.saveGameUrl(params[0])
                }
            }

            "saveAccountInfo" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.saveAccountInfo(params[0])
                }
            }

            "getAccountInfo" -> {
                result = mBridgeInterface.getAccountInfo()
            }

            "getAdjustDeviceID" -> {
                result = mBridgeInterface.getAdjustDeviceID()
            }

            "getGoogleADID" -> {
                result = mBridgeInterface.getGoogleADID()
            }

            "getIDFA" -> {
                result = mBridgeInterface.getIDFA()
            }

            "getReferID" -> {
                result = mBridgeInterface.getReferID()
            }

            "getAgentID" -> {
                result = mBridgeInterface.getAgentID()
            }

            "setCocosData" -> {
                if (params.size >= 2) {
                    mBridgeInterface.setCocosData(params[0], params[1])
                }
            }

            "getCocosData" -> {
                if (params.isNotEmpty()) {
                    result = mBridgeInterface.getCocosData(params[0])
                }
            }

            "getCocosAllData" -> {
                result = mBridgeInterface.getCocosAllData()
            }

            "getLighterHost" -> {
                result = mBridgeInterface.getLighterHost()
            }

            "getBridgeVersion" -> {
                result = mBridgeInterface.getBridgeVersion().toString()
            }

            "isFacebookEnable" -> {
                result = mBridgeInterface.isFacebookEnable().toString()
            }

            "getTDTargetCountry" -> {
                result = mBridgeInterface.getTDTargetCountry()
            }

            "openUrlByBrowser" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.openUrlByBrowser(params[0])
                }
            }

            "openUrlByWebView" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.openUrlByWebView(params[0])
                }
            }

            "openApp" -> {
                if (params.size >= 2) {
                    mBridgeInterface.openApp(params[0], params[1])
                }
            }

            "loadUrl" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.loadUrl(params[0])
                }
            }

            "goBack" -> {
                mBridgeInterface.goBack()
            }

            "close" -> {
                mBridgeInterface.close()
            }

            "refresh" -> {
                mBridgeInterface.refresh()
            }

            "clearCache" -> {
                mBridgeInterface.clearCache()
            }

            "saveImage" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.saveImage(params[0])
                }
            }

            "savePromotionMaterial" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.savePromotionMaterial(params[0])
                }
            }

            "synthesizePromotionImage" -> {
                if (params.size >= 4) {
                    mBridgeInterface.synthesizePromotionImage(
                        params[0],
                        parseInt(params[1]),
                        parseInt(params[2]),
                        parseInt(params[3])
                    )
                }
            }

            "shareUrl" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.shareUrl(params[0])
                }
            }

            "loginFacebook" -> {
                mBridgeInterface.loginFacebook()
            }

            "logoutFacebook" -> {
                mBridgeInterface.logoutFacebook()
            }

            "preloadPromotionImage" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.preloadPromotionImage(params[0])
                }
            }

            "shareToWhatsApp" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.shareToWhatsApp(params[0])
                }
            }

            "isHttpDnsEnable" -> {
                result = mBridgeInterface.isHttpDnsEnable().toString()
            }

            "httpdns" -> {
                if (params.isNotEmpty()) {
                    result = mBridgeInterface.httpdns(params[0])
                }
            }

            "httpdnsInit" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.httpdnsInit(params[0])
                }
            }

            "httpdnsRequestSync" -> {
                if (params.size >= 2) {
                    result = mBridgeInterface.httpdnsRequestSync(
                        params[0], stringToBytes(params[1])
                    )
                }
            }

            "httpdnsRequestAsync" -> {
                if (params.size >= 2) {
                    mBridgeInterface.httpdnsRequestAsync(params[0], stringToBytes(params[1]))
                }
            }

            "httpdnsWsOpen" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.httpdnsWsOpen(params[0])
                }
            }

            "httpdnsWsSend" -> {
                if (params.size >= 2) {
                    result = mBridgeInterface.httpdnsWsSend(params[0], stringToBytes(params[1]))
                }
            }

            "httpdnsWsClose" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.httpdnsWsClose(params[0])
                }
            }

            "httpdnsWsConnected" -> {
                if (params.isNotEmpty()) {
                    result = mBridgeInterface.httpdnsWsConnected(params[0])
                }
            }

            "getBuildVersion" -> {
                result = mBridgeInterface.getBuildVersion()
            }

            "onAnalysisStart" -> {
                if (params.size >= 2) {
                    mBridgeInterface.onAnalysisStart(params[0], parseLong(params[1]))
                }
            }

            "onAnalysisEnd" -> {
                mBridgeInterface.onAnalysisEnd()
            }

            "memoryInfo" -> {
                result = mBridgeInterface.memoryInfo()
            }

            "isEmulator" -> {
                result = mBridgeInterface.isEmulator().toString()
            }

            "commonData" -> {
                result = mBridgeInterface.commonData()
            }

            "exitApp" -> {
                mBridgeInterface.exitApp()
            }

            "handleNotification" -> {
                mBridgeInterface.handleNotification()
            }

            "onWebViewLoadChanged" -> {
                if (params.isNotEmpty()) {
                    mBridgeInterface.onWebViewLoadChanged(params[0])
                }
            }
        }
        return result
    }
}
